pkg/source/adapter/adapter.go

```diff
@@ -226,3 +232,71 @@ func (a *Adapter) Cleanup(sess sarama.ConsumerGroupSession) {
 		a.logger.Warnf("Cannot send the claims update: %v", err)
 	}
 }
+
+// InitOffsets makes sure all consumer group offsets are set.
+func (a *Adapter) InitOffsets(addrs []string, config *sarama.Config) func(sarama.ConsumerGroupSession) error {
+	return func(session sarama.ConsumerGroupSession) error {
+		// We want to make sure that ALL consumer group offsets are set to avoid
+		// losing events in case the consumer group session is closed before at least one message is
+		// consumed from ALL partitions.
+		// If not, an event sent to a partition with an uninitialized offset
+		// will not be forwarded when the session is closed (or a rebalancing is in progress).
+		kafkaClient, err := sarama.NewClient(addrs, config)
+		if err != nil {
+			return fmt.Errorf("failed to create a Kafka client: %w", err)
+		}
+		defer kafkaClient.Close()
+
+		kafkaAdminClient, err := sarama.NewClusterAdminFromClient(kafkaClient)
```

Comment:
The problem with this approach is that it makes the non-so-trivial assumption that the kafka config provided to access the kafka cluster has admin rights. IMO this assumption might not work in a lot of production use cases... Any ideas how to go around it? Or at least, make this behaviour optional?

Reply:
Now if the admin client cannot be created a warning is printed in the log. It's not an error anymore.